spec/calculator_spec.rb

```diff
@@ -1 +1,154 @@
-# Your code here
\ No newline at end of file
+# spec/calculator_spec.rb
+require 'calculator'
+describe Calculator do
+    let(:c){Calculator.new}
+
+    describe '#add' do
+        it 'returns the sum of its postive arguments' do
+              expect(c.add(1, 2)).to eq(3)
+        end
+        it 'returns the sum of its negative arguments' do
+            expect(c.add(-1, 2)).to eq(1)
+        end
+        it 'returns the sum of its float arguments' do
+            expect(c.add(0.1, 0.2)).to be_within(0.5).of(0.3)
+        end
+        it 'returns the sum of its integer and float arguments' do
+            expect(c.add(1, 0.2)).to eq(1.2)
+        end
+    end  #add
+
+    describe '#subtract' do
+        it 'returns the difference of its postive arguments' do
+            expect(c.subtract(1, 2)).to eq(-1)
+        end
+        it 'returns the difference of with a negative argument' do
+            expect(c.subtract(1, -2)).to eq(3)
+        end
+        it 'returns the different of its float arguments' do
+            expect(c.add(-0.1, -0.2)).to be_within(0.5).of(0.1)
+        end
+        it 'returns the difference of its float and integer arguments' do
+            expect(c.subtract(1, 0.2)).to eq(0.8)
+        end
+    end #subtract
+
+    describe '#multiply' do
+        it 'returns the product of its arguments' do
+            expect(c.multiply(2, 3)).to eq(6)
+        end
+        it 'returns a negative product for an odd number of negative arguments' do
+            expect(c.multiply(2, -3)).to eq(-6)
+        end
+        it 'returns the product of its float arguments' do
+            expect(c.multiply(0.2, 0.3)).to be_within(0.5).of(0.06)
+        end
+    end #multiply
+
+    describe '#divide' do
+        it 'returns the quotient of its arguments' do
+            expect(c.divide(2, 3)).to be_within(0.5).of(0.666)
+        end
+        it 'raises an ArgumentError if the second argument is zero' do
+            expect{c.divide(2, 0)}.to raise_error(ArgumentError)
+        end
+        it 'returns an integer if there is no remainder' do
+            expect(c.divide(4, 2)).to eq(2)
+        end
+        it 'returns a float if there is a remainder' do
+            expect(c.divide(2, 3)).to be_within(0.5).of(0.666)
```

Comment:
Thank you for all of these reviews.  I will work my way through them to improve my projects.  Completed.